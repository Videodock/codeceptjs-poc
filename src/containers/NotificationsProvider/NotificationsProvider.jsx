import React from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import { makeStyles } from '@material-ui/styles';
import Snackbar from '@material-ui/core/Snackbar';
// import Alert from '@material-ui/lab/Alert';
import * as PropTypes from 'prop-types';
import styles from './NotificationsProvider.styles';

const useStyles = makeStyles(styles);

export default function NotificationsProvider ({
  children,
  open,
  message,
  handleClose,
  type,
}) {
  const { t }   = useTranslation();
  const classes = useStyles();

  // onClose={handleClose} 
  // severity={type}

  return (
    <React.Fragment>
      {children}
      <Snackbar 
        open={open} 
        autoHideDuration={6000} 
        onClose={handleClose}
      >
        {/* <Alert severity="success"> */}
        {message}
        {/* </Alert> */}
      </Snackbar>
    </React.Fragment>
  );
}

NotificationsProvider.propTypes = {
  open       : PropTypes.bool,
  message    : PropTypes.string,
  handleClose: PropTypes.func,
  type       : PropTypes.string,
};

NotificationsProvider.defaultProps = {
  open: true,
  type: 'success',
};
