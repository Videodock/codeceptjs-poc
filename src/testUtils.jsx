import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
