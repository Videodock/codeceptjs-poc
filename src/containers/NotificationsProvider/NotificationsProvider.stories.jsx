import React from 'react';
import { storiesOf } from '@storybook/react';
import NotificationsProvider from './NotificationsProvider';

storiesOf('NotificationsProvider', module)
  .add('basic', () => <NotificationsProvider />);
