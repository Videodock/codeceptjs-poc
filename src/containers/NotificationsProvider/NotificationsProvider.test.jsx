import React from 'react';
import { render } from '../../testUtils';
import NotificationsProvider from './NotificationsProvider';

describe('<NotificationsProvider />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<NotificationsProvider />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});