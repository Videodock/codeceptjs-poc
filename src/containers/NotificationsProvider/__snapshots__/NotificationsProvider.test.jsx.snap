// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationsProvider /> renders and matches snapshot 1`] = `
<div
  class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
>
  <div
    class="MuiPaper-root MuiSnackbarContent-root MuiPaper-elevation6"
    direction="up"
    role="alert"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiSnackbarContent-message"
    />
  </div>
</div>
`;
